### DISCUSSION TITLE
Debugging/developing inside PyCharm (or other IDE)

### DISCUSSION THREAD

**Discussion by noam1023** [castorini/anserini#1699] [General]
I want to write code and debug it inside the IDE. A naive import of the MVN project into Pycharm fails to build (looks like lib path problems).

A specific example is adding/modifying a ranking function.

Also, even changing a single Java file triggers a 25 seconds build. Is this normal?

Is there a guide somewhere for newcomers? If not, I will write one (once I have a stable doc)
thanks
Noam

**Comment by lintool** [comment]
Hi @noam1023 - I use PyCharm but do most of my work in the shell... so any contributions here would be appreciated!
